Data/DynamicInputTexture.cs

```diff
@@ -1,152 +1,167 @@
-﻿using DolphinDynamicInputTextureCreator.Other;
-using Microsoft.Win32;
-using Newtonsoft.Json;
+﻿using Newtonsoft.Json;
 using System;
 using System.Collections.ObjectModel;
 using System.Drawing;
 using System.IO;
 using System.Linq;
-using System.Windows.Input;
 
 namespace DolphinDynamicInputTextureCreator.Data
 {
     [JsonObject(IsReference = true)]
-    public class DynamicInputTexture : Other.PropertyChangedBase
+    public class DynamicInputTexture : Other.PropertyChangedBase, Interfaces.IExportableImage
     {
+
         #region PROPERTIES
+
         /// <summary>
-        /// The hash of the texture, as defined by Dolphin
+        /// The path to the texture before any input modification
         /// </summary>
-        private string _texture_hash;
-        public string TextureHash
+        public string TexturePath
         {
-            get
-            {
-                return _texture_hash;
-            }
+            get => _texture_path;
             set
             {
-                _texture_hash = value;
-                OnPropertyChanged(nameof(TextureHash));
+                _texture_path = value;
+                TextureHash ??= Path.GetFileName(value);
+                OnPropertyChanged(nameof(TexturePath));
+                UpdateImageWidthHeight();
             }
         }
+        private string _texture_path;
 
         /// <summary>
-        /// The path to the texture before any input modification
+        /// if the export path should not be calculated.
         /// </summary>
-        private string _texture_path;
-        public string TexturePath
+        public string RelativeExportPath
```

Comment:
I'm not sure I see the value in this being a property on the class since it is used solely by the json writer.  Also the summary text makes it seem like a boolean.

Reply:
the RelativeExportPath is meant for packages that have been imported, because the packages work with Relative Path and I want to keep the original data. Another possibility would be to save only the root directory, so that we can generate the relative paht again.